npm/pkg/dataplane/policies/policy.go

```diff
@@ -110,7 +111,6 @@ func (aclPolicy *ACLPolicy) hasKnownProtocol() bool {
 	return aclPolicy.Protocol != "" && (aclPolicy.Protocol == TCP ||
 		aclPolicy.Protocol == UDP ||
 		aclPolicy.Protocol == SCTP ||
-		aclPolicy.Protocol == ICMP ||
```

Comment:
Is there a reason for us to not support ICMP ? Network policy rules can include ICMP as protocol right ?

Reply:
Thank you for bring it up. 
Confirmed it with [networkpolicy spec](https://github.com/kubernetes/api/blob/cf43cc39e22390e13043e176fdb787cfccdb1ef8/networking/v1/types.go#L140) and AKS cluster.

```
// The protocol (TCP, UDP, or SCTP) which traffic must match. If not specified, this
// field defaults to TCP.
```
```yaml
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: icmp-test
  namespace: default
spec:
  podSelector:
    matchLabels:
      role: db
  policyTypes:
  - Ingress
  ingress:
  - from:
    - ipBlock:
        cidr: 10.0.0.0/24
    ports:
    - protocol: ICMP
```
```shell
kubectl apply -f icmp.yaml
The NetworkPolicy "icmp-test" is invalid: spec.ingress[0].ports[0].protocol: Unsupported value: "ICMP": supported values: "TCP", "UDP", "SCTP"
```